envs/conda_requirements.yaml

```diff
@@ -4,26 +4,27 @@ channels:
   - bioconda
   - defaults
 dependencies:
-  - python=3.7
-  - snakemake=5.5.4
-  - biopython=1.74
-  - blast=2.9
-  - r=3.5.1
-  - r-getopt=1.20.2
-  - r-argparser=0.4
-  - r-futile.logger=1.4.3
-  - r-glue=1.3.0
-  - r-plyr=1.8.4
-  - r-dplyr=0.7.6
-  - r-tibble=2.1.3
-  - r-reshape2=1.4.3
-  - r-rcolorbrewer=1.1
-  - r-ggplot2=3.1.1
-  - r-ape=5.2
-  - r-maps=3.3.0
-  - r-phytools=0.6
-  - r-tidytree=0.2.5
-  - bioconductor-treeio=1.6.1
-  - bioconductor-ggtree=1.14.4
-  - r-gridextra=2.3
-  - r-egg=0.4.5
+  - python>=3.8
```

Comment:
You may actually want to lock in some of these dependency versions. This is especially true for dependencies that change quite rapidly and break APIs. I had run into a problem before where libraries like pandas were left ">=" in the dependency file. When installing on another computer, suddenly, the software no longer works because the newer version of pandas, which was installed by default, broke the code. You can also specify ranges in the requirements potentially. Like from 1.2 to 1.11 or something like that. For example, pandas 1.0 had considerable changes, and therefore, I had to set the dependency file to specify pandas >0.8 but less than 1.0. If you have tests and continuous integration set up, you can set up something like https://dependabot.com, which tests your code against new dependency versions to see if the unit or end-to-end tests fail with new versions. The service automatically performs a pull request with updated dependency files that specify the new dependency versions that pass the automated tests.

Reply:
@LeeBergstrand Good point - my hope was to keep the version numbers open ended to prevent the installation from breaking as easily in the future (some R packages can have very finicky conda installs), but it is probably better to lock down the version numbers, at least somewhat, to prevent future dependency changes from breaking the code.

I've now locked all version numbers to either major versions or to version ranges. End-to-end test still passes. Note that several of these packages failed to install when version ranges were specified, forcing me to lock them to major version numbers instead. Let me know if you think this is okay to merge and publish as `2.0.0-alpha5` now.